volmdlr/wires.py

```diff
@@ -188,6 +188,47 @@ def sort_points_along_wire(self, points):
 
         return sorted(points, key=lambda point: self.abscissa(point))
 
+    def is_ordered(self, tol=1e-6):
+        '''
+        check if the wire's primitives are ordered or not
+        '''
+
+        for primitive_1, primitive_2 in zip(self.primitives, self.primitives[1:]):
+            if primitive_1.end.point_distance(primitive_2.start) > tol:
+                return False
+        return True
+
+    def order_wire(self, tol=1e-6):
+        '''
+        order wire's primitives
+        '''
+
+        if self.is_ordered(tol=tol):
+            return self
+
+        new_primitives = [self.primitives[0]]
+        primitives = self.primitives[1:]
+        length_primitives = len(primitives)+1
+
+        while len(new_primitives)<length_primitives:
+            for primitive in primitives:
+                if new_primitives[0].start.point_distance(primitive.start) < tol:
+                    new_primitives.insert(0, primitive.reverse())
+                    primitives.remove(primitive)
+                elif new_primitives[-1].end.point_distance(primitive.start) < tol:
+                    new_primitives.append(primitive)
+                    primitives.remove(primitive)
+                elif new_primitives[0].start.point_distance(primitive.end) < tol:
+                    new_primitives.insert(0, primitive)
+                    primitives.remove(primitive)
+                elif new_primitives[-1].end.point_distance(primitive.end) < tol:
+                    new_primitives.append(primitive.reverse())
+                    primitives.remove(primitive)
+
+        self.primitives = new_primitives
+
+        return self
```

Comment:
better not to change data inplace but return a new instance

Reply:
It's done !